File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        # 
        # dp[i] = dp[i-1] + dp[i-2]
        # for 3 , steps to reach 2 + steps to reach 1 , 2 + 1 = 3 
        dp = [0,1]
        # dp[0] = 0
        # dp[1] = 1
        
        # 0 1 2
        # 1,1 1,2 2,3
        for i in range(n):
            temp = dp[1]
            
            dp[1] = dp[1] + dp[0]
            dp[0] = temp
            
        return dp[1]

File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        int ways1 = 0;
        int ways2 = 1;

        for (int i =1;i<=n;i++) {      
            int temp = ways2;
            ways2 = ways2 + ways1;
            ways1 = temp; 
        }
        return ways2;
    }
}